src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,29 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(
+      UUID snapshotId, String tableName, int limit, int offset) {
+    Snapshot snapshot = retrieve(snapshotId);
+
+    snapshot.getTables().stream()
+        .filter(t -> t.getName().equals(tableName))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new SnapshotPreviewException(
+                    "No snapshot table exists with the name: " + tableName));
+
+    try {
+      List<Map<String, Object>> values =
+          bigQueryPdao.getSnapshotTable(snapshot, tableName, limit, offset);
+
+      return new SnapshotPreviewModel().result(new ArrayList<>(values));
+    } catch (InterruptedException e) {
+      throw new SnapshotPreviewException(
```

Comment:
I was reading about this recently and I now think it's an error to catch (and not rethrow) `InterruptedException` without interrupting the current thread. https://stackoverflow.com/a/3976377